src/include/common/container/producer_consumer_queue.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <utility>
+#include "common/macros.h"
+#include "spdlog/details/mpmc_blocking_q.h"
```

Comment:
spdlog is a logging library. This MPMC queue is its own implementation detail. For me, it seems not a good choice. TBB already provides two kinds of concurrent queues. The wrapper of one is already provided in https://github.com/cmu-db/terrier/blob/master/src/include/common/container/concurrent_queue.h. And our checkpoint group already implemented the wrapper for the other one in https://github.com/cmu-db/terrier/blob/dd70de38f0d649ab38937750aae5c735869f0215/src/include/common/container/concurrent_blocking_queue.h. I hope these will be helpful.

Reply:
We need both blocking and non-blocking dequeue. That's why we resorted to MPMC

Reply:
I don't think this is a good approach.

I do not believe that the internals (i.e. the details directory) of spdlog is intended to be public. The documentation and examples use the spdlog.h header file along with  the various loggers as the public API. There is no promise that some future version of spdlog won't replace the internals.

So, possible choices that come to mind:
- Use one of the queue libraries e.g. the TBB or similar. This would be preferred.
- If you must use MPMC, then place a copy in a separate third party directory and use it from there. Add some basic units tests. You do not want to be dependent on the internal components of packages.